pkg/build/api/v1beta3/types.go

```diff
@@ -85,6 +85,9 @@ type BuildStatus struct {
 
 	// Config is an ObjectReference to the BuildConfig this Build is based on.
 	Config *kapi.ObjectReference `json:"config,omitempty"`
+
+	// Revision is the revision that was built with this build
+	Revision *SourceRevision `json:"revision,omitempty" description:"source revision built with this build"`
 }
```

Comment:
The Revision struct won't get set with a branch ref... that's what the Source.Ref is for. We currently only set it to a specific commit when receiving a post in the webhook. Had a discussion about it with @bparees and I think we agreed that it's technically part of the spec. We're just completing it when we actually get to download the code. Whenever you clone a particular build, we'll build the same commit that was built before.